code/controllers/subsystem/mapping.dm

```diff
@@ -184,17 +184,23 @@ SUBSYSTEM_DEF(mapping)
 				if(index)
 					lists_to_reserve.Cut(1, index)
 				return
-			var/turf/T = packet[packetlen]
-			T.empty(RESERVED_TURF_TYPE, RESERVED_TURF_TYPE, null, TRUE)
-			LAZYINITLIST(unused_turfs["[T.z]"])
-			unused_turfs["[T.z]"] |= T
-			var/area/old_area = T.loc
-			LISTASSERTLEN(old_area.turfs_to_uncontain_by_zlevel, T.z, list())
-			old_area.turfs_to_uncontain_by_zlevel[T.z] += T
-			T.turf_flags = UNUSED_RESERVATION_TURF
-			world_contents += T
-			LISTASSERTLEN(world_turf_contents_by_z, T.z, list())
-			world_turf_contents_by_z[T.z] += T
+			var/turf/reserving_turf = packet[packetlen]
+			reserving_turf.empty(RESERVED_TURF_TYPE, RESERVED_TURF_TYPE, null, TRUE)
+			LAZYINITLIST(unused_turfs["[reserving_turf.z]"])
+			unused_turfs["[reserving_turf.z]"] |= reserving_turf
+			var/area/old_area = reserving_turf.loc
+			LISTASSERTLEN(old_area.turfs_to_uncontain_by_zlevel, reserving_turf.z, list())
+			old_area.turfs_to_uncontain_by_zlevel[reserving_turf.z] += reserving_turf
+			reserving_turf.turf_flags = UNUSED_RESERVATION_TURF
+
+			reserving_turf.blocks_air = TRUE // reservation turfs are not allowed to interact with atmos at all
+			for(var/turf/turf_neighbor as anything in reserving_turf.atmos_adjacent_turfs)
+				turf_neighbor.atmos_adjacent_turfs -= reserving_turf
+			reserving_turf.atmos_adjacent_turfs.Cut()
```

Comment:
this code is redundant. the turf in question is queued to have its atmos adjacent turfs rebuild regardless by empty() (which should be commented here) which will force blocks_air to come into effect. you should not be fucking with these lists.